uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-assignment.controller.js

```diff
@@ -259,6 +261,91 @@ define([
       });
     }
 
+    /**
+     * Returns the default assignee for the activity type provided.
+     *
+     * @param {Object} activityType the activity type definition.
+     * @return {Promise} resolves to an array wrapping the default assignee's contact id or NULL in case of none.
+     */
+    function getDefaultAssigneeForActivityType (activityType) {
+      switch (activityType.default_assignee_type) {
+        case defaultAssigneeOptionsIndex.SPECIFIC_CONTACT:
+          return $q.resolve([ activityType.default_assignee_contact ]);
+        case defaultAssigneeOptionsIndex.USER_CREATING_THE_CASE:
+          return $q.resolve([ session.contactId ]);
+        case defaultAssigneeOptionsIndex.BY_RELATIONSHIP:
+          return getDefaultAssigneeForActivityTypeByRelationship(activityType);
+        case defaultAssigneeOptionsIndex.NONE:
+        default:
+          return $q.resolve(null);
+      }
+    }
+
+    /**
+     * Returns the default assignee depending on the relationship to the selected
+     * target contact.
+     *
+     * @param {Object} activityType the activity type definition.
+     * @return {Promise}
+     */
+    function getDefaultAssigneeForActivityTypeByRelationship (activityType) {
+      // skip if a target contact has not been selected:
+      if (!$scope.assignment.client_id) {
+        return $q.resolve(null);
+      }
+
+      return Relationship.allValid({
+        'contact_id_a': $scope.assignment.client_id,
+        'relationship_type_id.name_b_a': activityType.default_assignee_relationship,
+        'options': { 'limit': 1 }
+      })
+        .then(function (result) {
+          return result.list.map(function (relationship) {
+            return relationship.contact_id_b;
+          });
+        });
+    }
+
+    /**
+     * Initializes the default assignee for each one of the tasks available.
+     */
+    function initDefaultAssigneesForTasks () {
+      // skip if an activity set and types have not been defined:
+      if (!$scope.activity.activitySet.activityTypes) {
+        return;
+      }
+
+      $scope.activity.activitySet.activityTypes.forEach(function (activityType) {
+        var task = _.find($scope.taskList, function (task) {
```

Comment:
Do not define variables inside loop. Please keep the `var task` part outside.

Reply:
The loop is inside a function. I did something similar before and @AkA84 corrected me: https://github.com/civicrm/civihr/pull/2462/files/071e2b05952e6824be9adf9ee157d003a61f2dc2#r167195203

I like his suggestion. Basically we have to define variables at the top of their function.

Reply:
Thats a good one.